bin/writer.rpi3

```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 Google LLC. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+# Writes rpi3_b to the board
+
+# Args:
+#  $1: Mount point of the sdcard when board is off
+#  $2:
+
+set -ex
+
+mount=$1
+build=$2
+
+echo "Writing to ${mount} from build at ${build}"
+
+# First make a copy of the old files if we haven't already
+if [[ ! -e ${mount}/config.old ]]; then
+    cp ${mount}/config.txt ${mount}/config.old
+fi
+if [[ ! -e ${mount}/rpi3-u-boot.bin.old ]]; then
+    cp ${mount}/rpi3-u-boot.bin ${mount}/rpi3-u-boot.bin.old
+fi
+
+# Enable the UART and fix the GPU frequency so it works correctly
+sed -i '/enable_uart/c\enable_uart = 1' /media/rpi3_b_boot/config.txt
+if ! grep -q gpu_freq=250 /media/rpi3_b_boot/config.txt; then
+    echo 'gpu_freq=250' >>/media/rpi3_b_boot/config.txt
+fi
+
+# Copy U-Boot over from the build directory
+cp ${build}/u-boot.bin ${mount}/rpi3-u-boot.bin
+
+sync
```

Comment:
This sync shouldn't be necessary because umount is syncrhonous, and is also duplicated in the script that calls/includes this script.